using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Demo
{
	class ThrowDemo
	{
		//static void ProcessString(string s)
		//{
		//	if (s == null)
		//	{
		//		throw new ArgumentNullException();
		//	}
		//}

		//static void Main()
		//{
		//	try
		//	{
		//		string s = null;
		//		ProcessString(s);
		//	}

			

		//	catch (ArgumentNullException e)
		//	{
		//		Console.WriteLine("{0} First exception caught.", e);
		//	}
		//	// if the below block is placed before the first catch block, error will appear: A previous catch clasue already catches....
		//	catch (Exception e)
		//	{
		//		Console.WriteLine("{0} second exception caught.", e);
		//	}

		//}


	}
}
